lib/internal/streams/operators.js

```diff
@@ -0,0 +1,269 @@
+'use strict';
+
+const { AbortController } = require('internal/abort_controller');
+const { isIterable } = require('internal/streams/utils');
+const {
+  codes: {
+    ERR_INVALID_ARG_TYPE,
+    ERR_INVALID_ARG_VALUE,
+  }, AbortError,
+} = require('internal/errors');
+
+const {
+  NumberIsInteger,
+  Promise,
+  PromiseReject,
+  Symbol,
+} = primordials;
+
+const kEmpty = Symbol('kEmpty');
+
+async function * map(fn, options) {
+  if (typeof fn !== 'function') {
+    throw new ERR_INVALID_ARG_TYPE(
+      'fn', ['Function', 'AsyncFunction'], this);
+  }
+
+  let concurrency = 1;
+  if (options?.concurrency != null) {
+    concurrency = options.concurrency;
+  } else if (options != null && typeof options !== 'object') {
+    concurrency = options;
+  }
+
+  if (!NumberIsInteger(concurrency)) {
```

Comment:
In general JavaScript APIs usually gracefully convert numbers to integers  (like `setTimeout(fn, 103.5)` being allowed). Not sure if that matters in this case.